Sources/Sentry/SentryOutOfMemoryTracker.m

```diff
@@ -0,0 +1,179 @@
+                                             object:nil];
+
+    SentryFileManager *fileManager = [[[SentrySDK currentHub] getClient] fileManager];
+    SentryAppState *previousAppState = [fileManager readAppState];
+
+    // Is the current process being traced or not? If it is a debugger is attached.
+    bool isDebugging = self.crashAdapter.isBeingTraced;
+
+    SentryAppState *currentAppState =
+        [[SentryAppState alloc] initWithAppVersion:self.options.releaseName
+                                         osVersion:UIDevice.currentDevice.systemVersion
+                                       isDebugging:isDebugging];
+    [fileManager storeAppState:currentAppState];
+
+    // If there is no previous app state, we can't do anything.
+    if (nil == previousAppState) {
+        return;
+    }
+
+    // If the app version is different we assume it's an upgrade
+    if (![currentAppState.appVersion isEqualToString:previousAppState.appVersion]) {
+        return;
+    }
+
+    // The OS was upgraded
+    if (![currentAppState.osVersion isEqualToString:previousAppState.osVersion]) {
+        return;
+    }
+
+    // Restarting the app in development is a termination we can't catch and would falsely report
+    // OOMs.
+    if (previousAppState.isDebugging) {
+        return;
+    }
+
+    // The app was terminated normally
+    if (previousAppState.wasTerminated) {
+        return;
+    }
+
+    // The app crashed on the previous run. No OOM.
+    if (self.crashAdapter.crashedLastLaunch) {
+        return;
+    }
+
+    // Was the app in foreground/active ?
+    // If the app was in background we can't reliably tell if it was an OOM or not.
+    if (!previousAppState.isActive) {
+        return;
+    }
+
+    SentryEvent *event = [[SentryEvent alloc] initWithLevel:kSentryLevelFatal];
+    // Set to empty list so no breadcrumbs of the current scope are added
+    event.breadcrumbs = @[];
+
+    SentryException *exception =
+        [[SentryException alloc] initWithValue:SentryOutOfMemoryExceptionValue
+                                          type:SentryOutOfMemoryExceptionType];
+    SentryMechanism *mechanism =
+        [[SentryMechanism alloc] initWithType:SentryOutOfMemoryExceptionType];
+    mechanism.handled = @(NO);
+    exception.mechanism = mechanism;
+    event.exceptions = @[ exception ];
+
+    [SentrySDK captureCrashEvent:event];
+
+#else
+    [SentryLog logWithMessage:@"NO UIKit -> SentryOutOfMemoryTracker will not track OOM."
+                     andLevel:kSentryLevelInfo];
+    return;
+#endif
+}
+
+- (void)stop
+{
+#if SENTRY_HAS_UIKIT
+    [NSNotificationCenter.defaultCenter removeObserver:self];
+#endif
+}
+
+/**
+ * It is called when an App. is receiving events / It is in the foreground and when we receive a
+ * SentryHybridSdkDidBecomeActiveNotification.
+ */
+- (void)didBecomeActive
+{
+    // We don't know if the hybrid SDKs post the notification from a background thread, so we
```

Comment:
L: reading the hybrid SDKs here, it'd be nice to run a flutter or RN app with this version just to be sure that this implementation would work for them too ootb

Reply:
I'm definitely going to test that. As we release an alpha, I think it's fine to do this after merging this PR. In case it doesn't work out, we can easily fix it afterward.

Reply:
did it work?

Reply:
Didn't try it yet. Have to release the alpha first 😁